independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/client/maven/MavenRegistryClientFactory.java

```diff
@@ -161,7 +161,7 @@ public RegistryClient buildRegistryClient(RegistryConfig config) throws Registry
         }
 
         return new RegistryClientDispatcher(config, platformsResolver,
-                Boolean.TRUE.equals(config.getPlatforms().getExtensionCatalogsIncluded())
+                Boolean.TRUE.equals(platformsConfig == null ? Boolean.FALSE : platformsConfig.getExtensionCatalogsIncluded())
```

Comment:
```suggestion
                (platformsConfig == null || platformsConfig.getExtensionCatalogsIncluded())
```

Reply:
This will result in an NPE when `platformsConfig.getExtensionCatalogsIncluded() returns null.